### PR TITLE
Attempts to run CI when getting PRs from scala-steward

### PR DESCRIPTION
**Pull request by cquiroz**
Signed-off-by: Carlos Quiroz <cquiroz@gemini.edu>

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,7 +1,12 @@
 name: ci
 
 on:
-  push
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
```
Comment (rpiaggio) [inline .github/workflows/ci.yml:9]: Won't this disable CI if we attempt a PR on another branch? (I don't know if omitting this makes the default all of them or any of them)

Comment (cquiroz) [reply .github/workflows/ci.yml:9]: I think so. Do we ever do that, maybe for intermediate branches?

Comment (cquiroz) [reply .github/workflows/ci.yml:9]: Maybe this is a better config:
```
on: [push, pull_request]
```

File: .mergify.yml [hunk 2]
```diff
@@ -2,7 +2,7 @@
 pull_request_rules:
   - name: automatically merge scala-steward's PRs
     conditions:
-      - author=geminihlswadmin
+      - author=scala-steward
```
Comment (rpiaggio) [inline .mergify.yml:5]: Good catch. I missed this when switching back to Scala Steward.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (rpiaggio) [reply .github/workflows/ci.yml]: We have done "PR on top of PR".

**Pull request merged**